packages/amplication-server/src/core/team/team.service.ts

```diff
@@ -307,21 +445,74 @@ export class TeamService {
     });
 
     await this.analytics.trackWithContext({
-      event: EnumEventType.TeamRemoveRoles,
+      event: EnumEventType.TeamAssignmentRemoveRoles,
       properties: {
-        teamName: team.name,
+        teamName: teamAssignment.team.name,
         rolesRemoved: roleIds.length,
+        resourceId: teamAssignment.resourceId,
       },
     });
 
-    return updatedTeam;
+    return updatedTeamAssignment;
   }
 
-  async roles(teamId: string): Promise<Role[]> {
-    return this.prisma.team
+  async deleteTeamAssignment(
+    args: DeleteTeamAssignmentArgs
+  ): Promise<TeamAssignment> {
+    const {
+      where: { teamId, resourceId },
+    } = args;
+
+    //check if team is already assigned to the resource
+    const teamAssignment = await this.prisma.teamAssignment.delete({
+      where: {
+        // eslint-disable-next-line @typescript-eslint/naming-convention
+        teamId_resourceId: {
+          teamId,
+          resourceId,
+        },
+      },
+      include: {
+        team: true,
+      },
+    });
+
+    await this.analytics.trackWithContext({
+      event: EnumEventType.TeamDeleteAssignment,
+      properties: {
+        teamId: teamAssignment.teamId,
+        resourceId: teamAssignment.resourceId,
+      },
+    });
+
+    return teamAssignment;
+  }
+
+  async validateAssignmentResource(resourceId: string, user: User) {
+    const resource = await this.prisma.resource.findUnique({
+      where: {
+        id: resourceId,
+        deletedAt: null,
+        project: {
+          workspaceId: user.workspace.id,
+        },
+      },
+    });
+
+    if (!resource) {
+      throw new AmplicationError(INVALID_RESOURCE_ID);
+    }
+  }
+
+  async getTeamAssignmentRoles(args: WhereTeamAssignmentInput) {
+    return this.prisma.teamAssignment
       .findUnique({
         where: {
-          id: teamId,
+          // eslint-disable-next-line @typescript-eslint/naming-convention
+          teamId_resourceId: {
```

Comment:
i don't like to see this in js teamId_resourceId :) but i guess it's some convention/prisma thing so i'm ok with it